### PR TITLE
Proper inheritance for Events, and other fixes

### PR DESCRIPTION
**Pull request by ferm10n**
A handful of small modifications:
- I found that the Shape class and all the others that inherit from it were missing the functions given by Two.Utils.Events. I fixed this by having the Events object be extended to the Shape.prototype.
- I kinda missed having the .listenTo from Backbonejs, so I added it back.
- getRequestAnimationFrame now attempts to get the request function from "requestAnimationFrame", without any prefix. This was to eliminate warnings I was seeing in Chrome's console: _'webkitRequestAnimationFrame' is vendor-specific. Please use the standard 'requestAnimationFrame' instead_. If window.requestAnimationFrame isn't present then it'll use the old mechanism with vendor prefixes.
- Cloning a path was creating errors because I made clones before I added them to a scene or group. I added a check to ensure the parent is set before the clone is added to it.

### CODE REVIEW

File: src/effects/linear-gradient.js [hunk 1]
```diff
@@ -60,7 +60,8 @@
         clone[k] = this[k];
       }, this);
 
-      parent.add(clone);
+      if(parent)
+        parent.add(clone);
```
Comment (jonobr1) [inline src/effects/linear-gradient.js]: Mind wrapping these in brackets? e.g:
````javascript
if (parent) {
  parent.add(clone);
}
````

**Comment by jonobr1** [comment]
These are awesome updates! Thanks for contributing!

File: src/effects/linear-gradient.js (hunk 1 above)
Comment (ferm10n) [inline src/effects/linear-gradient.js]: Yup, can do haha

**Comment by ferm10n** [comment]
Thanks! Glad to help. Love the work done on this btw!
Feel free to check out what I was making with it: https://github.com/ferm10n/mitosis

**Pull request merged**